bindings/pydeck/pydeck/data_utils/type_checking.py

```diff
@@ -18,3 +7,21 @@ def is_pandas_df(obj):
         Returns True if object is a Pandas DataFrame and False otherwise
     """
     return obj.__class__.__module__ == "pandas.core.frame" and obj.to_records and obj.to_dict
+
+
+def has_geo_interface(obj):
+    return hasattr(obj, "__geo_interface__")
+
+
+def records_from_geo_interface(data):
+    """Un-nest data from object implementing __geo_interface__ standard"""
+    flattened_records = []
+    for d in data.__geo_interface__.get("features"):
+        record = {}
+        properties = d.get("properties")
+        geom = d.get("geometry")
+        for k in properties.keys():
+            record[k] = properties[k]
+            record["geometry"] = geom
```

Comment:
So here you're putting all properties at the top level instead of inside a `.properties` key?

For one, I'd love to see clean syntax like
```py
record = d.get('properties', {})
record['geometry'] = d.get('geometry')
```
(Not sure if you're ok with having `geometry` be `None`).

(Also here you're reassigning `record['geometry'] = geom` for every key, which I don't think you intend or need)

Second, I'm still worried about the user experience. For example there's a popular [`geojson`](https://github.com/jazzband/geojson) python package that wraps the json builtin and makes it easier to make compliant geojson features.

But the issue is that the objects in `geojson` "look" like GeoJSON `dict`s when printed, but they're _different_ objects provided by the `geojson` package. And these objects [have a `__geo_interface__` attribute](https://github.com/jazzband/geojson/blob/b8a8b8886a89907e13850f2807d081a23adc0b77/geojson/mapping.py#L14). So the user will pass an object that "looks" like GeoJSON but then won't be able to use `properties.key` in data driven styling because that key is now a top-level key.

Reply:
Beautiful use of `get`. Moments like these are why I love code reviews.

Unless there are some quick recommendations to address these situations, let's have a separate PR about supporting the edge case of `__geo_interface__`.